include/git2/remote.h

```diff
@@ -743,6 +743,20 @@ typedef struct {
 	 */
 	git_proxy_options proxy_opts;
 
+	/**
+	 * Depth of the fetch to perform, has to be a positive integer.
+	 *
+	 * The default is -1, which will fetch the full history.
```

Comment:
You state it has to be a positive integer but then specify that `-1` fetches the full history. The logic also reflects that `n < 1` fetches the full history. Could be a improved a bit. Perhaps...

```suggestion
	 * Depth of the fetch to perform.
	 * 
	 * depth >= 1 specifies depth; depth <= 0 fetches the full history.
	 *
	 * The default is -1.
```

Reply:
I think the original thought was that specifying "0" is really an input error, but then we decided to push that sort of error handling up into our client code for simplicity. Depth <= 0 corresponding to a full fetch would definitely be easier to reason about, further opinions welcome!